### PR TITLE
refactor(core): remove the automatic URL change feature

### PR DESCRIPTION
**Pull request by JimmFly**
close [AF-954](https://linear.app/affine-design/issue/AF-954/center-peek-中-editor-selection-变化不应修改-location-hash)

Since the URL changes constantly with the selection, causing the editor to render frequently and also affecting the use of center peek, we decided to remove the function of automatically changing the URL.

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7339** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7339?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by forehalo — COMMENTED** [review]
any context?

File: packages/frontend/core/src/hooks/affine/use-share-url.ts [hunk 1]
```diff
@@ -21,9 +22,8 @@ const useGenerateUrl = ({ workspaceId, pageId, urlType }: UseSharingUrl) => {
   // to generate a public url like https://app.affine.app/share/123/456
```
Comment (pengx17) [inline packages/frontend/core/src/hooks/affine/use-share-url.ts:27]: <i class='graphite__hidden'>[Re: line 18]<br/><br/></i>

This hook is used for the sharing panel and it seems feature-wise we do not need to sync the blockid into hash there.

Some implementation issues:
Consdering hash is the same with blockId, which should also be part of the `UseSharingUrl` props. Also, the given active editor may not necessarily the same with the given pageId.
Thus the hook may be refactored to a plain function.<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7339?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Comment by JimmFly** [comment]
> Since the URL changes constantly with the selection, causing the editor to render frequently and also affecting the use of center peek, we decided to remove the function of automatically changing the URL.

**Comment by JimmFly** [comment]
> any context?

Since the URL changes constantly with the selection, causing the editor to render frequently and also affecting the use of center peek, we decided to remove the function of automatically changing the URL. @forehalo

**Comment by forehalo** [comment]
> > any context?
> 
> Since the URL changes constantly with the selection, causing the editor to render frequently and also affecting the use of center peek, we decided to remove the function of automatically changing the URL. @forehalo 

cool

File: packages/frontend/core/src/hooks/affine/use-share-url.ts (hunk 1 above)
Comment (JimmFly) [reply packages/frontend/core/src/hooks/affine/use-share-url.ts:27]: I made some changes according to your suggestions. What do you think about this?

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/hooks/affine/use-share-url.ts [hunk 2]
```diff
@@ -73,6 +79,30 @@ export const useSharingUrl = ({
...
+    if (currentBlockSelection) {
+      setBlockId(`#${currentBlockSelection.blockId}`);
+    }
+
+    disposable = selectManager.slots.changed.on(selections => {
+      if (selections[0] && selections[0].type === 'block') {
+        setBlockId(`#${selections[0].blockId}`);
+      } else if (blockId.length > 0) {
+        setBlockId('');
+      }
```
Comment (EYHN) [inline packages/frontend/core/src/hooks/affine/use-share-url.ts]:
```suggestion
      setBlockId(
          (prev) => {
                if (selections[0] && selections[0].type === 'block') {
                  return `#${selections[0].blockId}`;
                 } else if (prev.length > 0) {
                  return '';
                 } else {
                   return prev;
                 }
          }
      )
```

File: packages/frontend/core/src/hooks/affine/use-share-url.ts [hunk 3]
```diff
@@ -73,6 +79,30 @@ export const useSharingUrl = ({
...
+    });
+    return () => {
+      disposable?.dispose();
+    };
+  }, [blockId.length, editor?.host?.selection, urlType]);
```
Comment (EYHN) [inline packages/frontend/core/src/hooks/affine/use-share-url.ts]:
```suggestion
  }, [editor?.host?.selection, urlType]);
```

**Review by EYHN — APPROVED** [review]

**Pull request merged**